contracts/QuestFactory.sol

```diff
@@ -549,20 +602,33 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
         ++currentQuest.numberMinted;
         questContract_.singleClaim(msg.sender);
 
-        if(mintFee > 0) processMintFee();
+        if(mintFee > 0) processMintFee(ref_);
 
         emit Quest1155Claimed(msg.sender, currentQuest.questAddress, questId_, questContract_.rewardToken(), questContract_.tokenId());
+
+        if(ref_ != address(0)) {
+            emit QuestClaimedReferred(msg.sender, currentQuest.questAddress, questId_, questContract_.rewardToken(), questContract_.tokenId(), ref_, referralFee);
+        }
+    }
+
+    function processMintFee(address ref_) private {
+        processExtraFee();
+        if (ref_ == address(0)) {
+            getMintFeeRecipient().safeTransferETH(mintFee);
+            return;
+        }
+        uint referralAmount = (mintFee * referralFee) / 10_000;
+        ref_.safeTransferETH(referralAmount);
+        getMintFeeRecipient().safeTransferETH(mintFee - referralAmount);
     }
 
-    function processMintFee() private {
+    function processExtraFee() private {
```

Comment:
[nit] extra fees sound like it's going to be actual extra fees lol. Maybe wondering if we can change to `processChange` or `processRefund` or something